holidays/holiday_groups.py

```diff
@@ -469,124 +469,322 @@ def _add_whit_sunday(self, name) -> date:
+            dt,
+        )
+
+    def _add_vesak(self, name) -> Optional[date]:
+        """
+        Add Vesak (15th day of the 4th lunar month).
+
+        Vesak for Thailand, Laos, Singapore and Indonesia.
+        https://en.wikipedia.org/wiki/Vesak
+        """
+        return self._add_buddhist_calendar_holiday(
+            name,
+            self._buddhist_calendar.vesak_date(self._year),
+        )
+
+    def _add_vesak_may(self, name) -> Optional[date]:
+        """
+        Add Vesak (on the day of the first full moon in May
+        in the Gregorian calendar).
+
+        Vesak for Sri Lanka, Nepal, India, Bangladesh and Malaysia.
+        https://en.wikipedia.org/wiki/Vesak
+        """
+        return self._add_buddhist_calendar_holiday(
+            name,
+            self._buddhist_calendar.vesak_may_date(self._year),
+        )
+
+
 class ChineseCalendarHolidays:
     """
     Chinese lunisolar calendar holidays.
     """
 
-    def __init__(self) -> None:
-        self._chinese_calendar = _ChineseLuniSolar()
+    def __init__(self, calendar=_ChineseLunisolar()) -> None:
+        self._chinese_calendar = calendar
 
     @property
     def _chinese_new_year(self):
         """
         Return Chinese New Year date.
         """
-        return self._chinese_calendar.lunar_n_y_date(self._year)
+        return self._chinese_calendar.lunar_new_year_date(self._year)[0]
 
-    def _add_dragon_boat_festival(self, name) -> date:
+    @property
+    def _mid_autumn_festival(self):
         """
-        Add Dragon Boat Festival (5th day of 5th lunar month).
-
-        The Dragon Boat Festival is a traditional Chinese holiday which occurs
-        on the fifth day of the fifth month of the Chinese calendar.
-        https://en.wikipedia.org/wiki/Dragon_Boat_Festival
+        Return Mid Autumn Festival (15th day of the 8th lunar month) date.
         """
-        return self._add_chinese_calendar_holiday(name, 5, 5)
+        return self._chinese_calendar.mid_autumn_date(self._year)[0]
 
-    def _add_chinese_calendar_holiday(self, name, month, day) -> date:
+    def _add_chinese_calendar_holiday(
+        self,
+        name: str,
+        hol_date: Tuple[date, bool],
+        days_delta: int = 0,
+    ) -> Optional[date]:
         """
-        Add Chinese lunar calendar holiday.
+        Add Chinese calendar holiday.
+
+        Appends customizable estimation label at the end of holiday name if
+        holiday date is an estimation.
         """
+        estimated_label = getattr(self, "estimated_label", "estimated")
+        show_estimated = getattr(self, "show_estimated", False)
+        dt, is_estimated = hol_date
+        if days_delta != 0:
+            dt += td(days=days_delta)
         return self._add_holiday(
+            f"{name}* (*{estimated_label})"
+            if is_estimated and show_estimated
+            else name,
+            dt,
+        )
+
+    def _add_chinese_buddha_birthday(self, name) -> Optional[date]:
```

Comment:
```suggestion
    def _add_chinese_birthday_of_buddha(self, name) -> Optional[date]:
```
sounds better to my ear, please consider renaming.

Reply:
Done.